HorizonEngine/src/HorizonEngine/SceneManagement/Scene.cpp

```diff
@@ -71,7 +72,8 @@ namespace Hzn
 			RigidBody2DComponent,
 			BoxCollider2DComponent,
 			RenderComponent,
-			CameraComponent>(outputArchive);
+			CameraComponent,
+			ScriptComponent>(outputArchive);
```

Comment:
good to add script component

Reply:
Yeah, this now serializes the script component as well.